File: frontend/src/api/feedback.ts
import api from './client';

export async function submitFeedback(articleId: string, feedbackType: 'show_more' | 'show_less' | 'irrelevant') {
  const { data } = await api.post('/feedback', { article_id: articleId, feedback_type: feedbackType });
  return data;
}

File: backend/app/schemas/feedback.py
from pydantic import BaseModel
from typing import Literal


class FeedbackCreate(BaseModel):
    article_id: str
    feedback_type: Literal["show_more", "show_less", "irrelevant"]


class FeedbackResponse(BaseModel):
    id: str
    article_id: str
    feedback_type: str
    message: str = "Feedback recorded"